clang/test/CheckedCRewriter/realloc.c

```diff
@@ -0,0 +1,19 @@
+// RUN: cconv-standalone -alltypes %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: cconv-standalone %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: cconv-standalone -output-postfix=checkedNOALL %s
+// RUN: %clang -c %S/realloc.checkedNOALL.c
+// RUN: rm %S/realloc.checkedNOALL.c
+
+
+#define size_t int
+extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);
+extern _Itype_for_any(T) void *realloc(void *pointer : itype(_Array_ptr<T>) byte_count(1), size_t size) : itype(_Array_ptr<T>) byte_count(size);
+
+void foo(int *w) { 
+    int *y = malloc(2*sizeof(int)); 
+    y[1] = 3;
+    int *z = realloc(y, 5*sizeof(int)); 
+    z[3] =  2;
+} 
+//CHECK: int *y = malloc(sizeof(int));
```

Comment:
Why are you not doing CHECK_ALL and CHECK_NOALL? Don't we want to see the `-alltypes` output too? Then you would also see that you are getting array types.

Reply:
It's the same for both -alltypes and non alltypes, so you'll see at the very top that I run it both with and without, but both tests look at `CHECK` annotations in addition to `CHECK_NOALL` and `CHECK_ALL`. Thought itd be easier this way so I wouldnt have to copy paste the same thing twice with two annotations.

Reply:
Then this is probably not the right fix. :( It should be an array pointer for both. Perhaps my suggestion below will solve it.